datumaro/plugins/splitter.py

```diff
@@ -70,21 +74,29 @@ def _get_uniq_annotations(dataset):
         return annotations
 
     @staticmethod
-    def _validate_splits(splits, valid=None):
+    def _validate_splits(splits, restrict=False):
         snames = []
         ratios = []
-        if valid is None:
-            valid = ["train", "val", "test"]
+        subsets = set()
+        valid = ["train", "val", "test"]
+        # remove subset name restriction
+        # regarding https://github.com/openvinotoolkit/datumaro/issues/194
         for subset, ratio in splits:
-            assert subset in valid, \
-                "Subset name must be one of %s, but got %s" % (valid, subset)
+            if restrict:
```

Comment:
What is the point of this restriction? Is it possible to require more than 3 subsets? Do subsets have special processing depending on the name?

Reply:
if restrict is Ture, the subset names are restricted into ["train", "val", "test"]. This is for re-identification at the moment.
For detection and classification, this restriction isn't needed so I changed this code. This is regarding issue #194